src/concepts/kafka-proxies/grpc-proxy.md

```diff
@@ -1,297 +1,51 @@
 ---
-description: This guide will walk through each unique gRPC message request and response design and how Zilla is configured to manage the connection for each.
+description: The Zilla gRPC Kafka Proxy lets you implement gRPC service definitions from protobuf files to consume and produce messages from Kafka topics.
 prev: false
 next: /tutorials/grpc/grpc-intro.md
 ---
 
 # gRPC Kafka Proxy
 
-This guide will walk through each unique gRPC message request and response design and how Zilla is configured to manage the connection for each.
+The Zilla gRPC Kafka Proxy lets you implement gRPC service definitions from protobuf files to consume and produce messages from Kafka topics.
 
-Let's take a look at how Zilla would be configured with a full featured gRPC service. For this, we will use the [route_guide.proto](https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/route_guide.proto) to define the gRPC service and the method request-response types. You can find examples of running this service in any language on the [gRPC docs](https://grpc.io/docs/languages/) as well as example clients like this one implemented in [java](https://grpc.io/docs/languages/java/basics/#client).
+A service method's request and response messages are routed on and off of Kafka. Zilla can be the server delivering messages to a Kafka topic or fanout messages from a topic to running gRPC services. Additionally, Zilla can sit on the critical path between a gRPC client and server. They can communicate as if they are talking to each other, with Zilla proxying the messages through Kafka.
 
-## Step 1: Declaring the service
+## Correlated Request-Response
 
-Zilla lets you use your service proto file(s) to specify the service definitions. This will let you create one or many [grpc-kafka](../../reference/config/bindings/binding-grpc-kafka.md) proxy handlers to be used by the whole service or individual methods.
+Zilla manages the synchronous request and response messages of a gRPC service on the event stream. Each message is correlated to each other with a `zilla:correlation-id` header, providing an identifier for both Zilla and Kafka workflows to act on. Correlated messages can be on the same or different Kafka topics.
```

Comment:
The concept of event stream in Kafka is typically used for facts (events) that have already happened, so this is better aligned terminology for fanout, whereas request-response is more like command-reply pattern than events, since commands are a request to do work and can be rejected.

> Each message is correlated to each other with a ...

Request and responses messages are correlated by a ...

> Correlated messages can be on the same or different Kafka topics.

Correlated messages can really only be on different topics.
The only exception to this is doing some kinds of echo test where the request is treated immediately as the response, and is therefore implicitly correlated because it is the same message with the same correlation header value.
In practice this is not useful, so we should only emphasize the need for 2 topics here, one for requests and one for responses.